### PR TITLE
Add Helpful Message For OSError

### PR DESCRIPTION
**Pull request by sibi361**
Suggest running the script on a different bind address or port using the -b or -p flags respectively

### CODE REVIEW

File: dns-mitm.py [hunk 1]
```diff
@@ -148,6 +148,10 @@ def init_listener():
         print("This script must be run as root if you want to listen " +
             "on port %d" % args.PORT)
         exit(1)
+    except OSError as e:
+        if re.match(".*Address already in use.*", str(e)):
+            print("Address already in use: try running on a different bind address/port - see help (-h)")
+            exit(0)
```
Comment (AdrianVollmer) [inline dns-mitm.py:154]: Can you add these two lines here?

```
else:
    raise
````

Comment (sibi361) [reply dns-mitm.py:154]: I've added them, but what exactly is the use of this clause here?

Comment (AdrianVollmer) [reply dns-mitm.py:154]: Thanks!

Without it, exceptions of type `OSError`  that don't match your regex would simply be swallowed and lead to even more confusing errors.

**Comment by sibi361** [comment]
@AdrianVollmer sorry to bother you about this again, but I think I made a mistake in 7b31ec4

I had added the `else: raise` under the try-except block:
```
except OSError as e:
    if re.match(".*Address already in use.*", str(e)):
        print("Address already in use: try running on a different bind address/port - see help (-h)")
        exit(0)
else:
    raise
```

instead of:

```
except OSError as e:
    if re.match(".*Address already in use.*", str(e)):
        print("Address already in use: try running on a different bind address/port - see help (-h)")
        exit(0)
    else:
        raise
```

Should I create another PR to fix this?

**Comment by AdrianVollmer** [comment]
sure, go ahead. thanks

**Pull request merged**